pkg/module/parse/template_to_module.go

```diff
@@ -134,6 +160,28 @@ func NewManifestFromTemplate(
 	return manifest, nil
 }
 
+func getRemoteDescriptor(
+	ctx context.Context,
+	template *v1beta1.ModuleTemplate,
+	descriptor *v1beta1.Descriptor,
+	cache *ocmextensions.ComponentDescriptorCache,
+	clnt client.Client,
+) (*compdesc.ComponentDescriptor, error) {
+	descriptorCacheKey, err := template.GetComponentDescriptorCacheKey()
+	if err != nil {
+		return nil, err
+	}
+	remoteDescriptor := cache.Get(descriptorCacheKey)
```

Comment:
This sequence: `(Get() -> (If nil then -> GetRemoteDescriptor() -> Set())` looks like it will be a common pattern. Perhaps it's worth to add this as a public method in the cache?

Reply:
good point!